README.md

```diff
@@ -1,4 +1,199 @@
-    composer require efabrica/nette-database-repository
+Bring better static analysis to your Nette Database! Typehinted entities (ActiveRows), queries (Selections) and repositories (services). 
+
+## Installation
+
+The best way to install this extension is using [Composer](http://getcomposer.org/):
+
+```sh
+$ composer require efabrica/nette-repository
+```
+
+Register our configuration:
+```neon
+includes:
+    - %appDir%/../vendor/efabrica/nette-repository/core/config.neon
+    
+symfonyConsoleApp:
+    type: Symfony\Component\Console\Application
+    setup:
+        - add(@repoCodeGen)
+```
+
+Execute our repository code generation command:
+```sh
+$ php bin/console repository:code-gen
+```
+
+## Usage
+
+### Entity
+
+The `Entity` class extends `ActiveRow` but is made to be used as a base class for your entities. 
+It is supposed to be generated by our code generation tool, but you can also write it manually.
+
+#### Inserting an entity
+
+```php
+assert($repository instanceof PersonRepository);
+assert($entity instanceof Person);
+$entity = $repository->createRow();
+$entity->name = 'John';
+$entity->surname = 'Doe';
+$entity->age = 42;
+$entity->save();
+```
+
+Classic:
+```php
+$person = $repository->insert([
+    Person::name => 'John',
+    Person::surname => 'Doe',
+    Person::age => 42,
+]);
+```
+
+Multi-insert:
+```php
+$persons = [];
+foreach (range(30, 40) as $age) {
+    $person = $repository->createRow();
+    $person->name = 'John';
+    $person->surname = 'Doe';
+    $person->age = $age;
+    $persons[] = $person;
+}
+$repository->insert($persons);
+```
+
+#### Updating an entity
+
+```php
+$entity = $repository->find($id);
+$entity->name = 'Jake';
+$entity->save();
+```
+
+Classic:
+```php
+$repository->update($id, [
+    Person::name => 'Jake',
+]);
+```
+
+#### Deleting an entity
+
+```php
+$entity = $repository->find($id);
+$entity->delete();
+```
+
+Or:
+```php
+$repository->delete($id);
+```
+### Scopes
+Scope is a way to define which behaviors are enabled for repository, query and entity recursively.
+
+Recursively means, that if you define a scope for repository, it will be applied to query and entity as well.
+
+`->scopeRaw()` returns a clone of the object with raw scope. Raw scope removes all behaviors.
+
+`->scopeFull()` returns a clone of the object with full scope. Full scope keeps all behaviors.
+
+#### Example
+```php
+final class ApiScope implements \Efabrica\NetteRepository\Repository\Scope\Scope
+{
+    public function apply(RepositoryBehaviors $behaviors, Repository $repository): void
+    {
+        $behaviors
+            ->remove(SoftDeleteBehavior::class)
+            ->remove(PublishBehavior::class);
+        // You should only remove behaviors, as the scope can be applied on different repositories at once.
+        
+        // If you had a user repository and injected the user into the scope, you could do this:
+        if ($repository instanceof UserRepository && $this->user->isAdmin()) {
```

Comment:
Is this a good example with User? What other use cases we can think of? Adding a behavior based on some configuration setting value?

Reply:
Yes, a container parameter sounds like a better example. But you can do anything since Scope can be a service.